any_parser/async_parser.py

```diff
@@ -0,0 +1,58 @@
+"""Asynchronous parser implementation."""
+
+import json
+from pathlib import Path
+from typing import Dict, Optional
+
+import requests
+
+from any_parser.base_parser import BaseParser, ProcessType
+from any_parser.utils import upload_file_to_presigned_url
+
+TIMEOUT = 60
+
+
+class AsyncParser(BaseParser):
+    def _setup_endpoints(self) -> None:
+        self._async_upload_url = f"{self._base_url}/async/upload"
+        self._async_fetch_url = f"{self._base_url}/async/fetch"
+
+    def send_async_request(
```

Comment:
Another big problem of OOP is suddenly introduce new method which is not shown in the base class. Python is a very tolerable language, but we should still follow the best standard. 

For example, in the BaseParser, there is neither `send_async_request` for the async and `get_sync_response` for the sync. I would suggest you to add a `parse` method for BaseParser first and raise NotImplemented exception first. Then, you should implement it for both sync and async. This will only only make our code more OOP, but also improved the readability.